packages/services/src/vckit.service.ts

```diff
@@ -43,18 +51,58 @@ export const issueVC = async ({
   type,
   issuer,
   credentialSubject,
+  credentialStatus,
   restOfVC,
   vcKitAPIUrl,
 }: IVcKitIssueVC): Promise<VerifiableCredential> => {
-  const body = constructCredentialObject({ context, type, issuer, credentialSubject, ...restOfVC });
-  privateAPI.setBearerTokenAuthorizationHeaders(appConfig.defaultVerificationServiceLink.apiKey ?? '');
+  const apiKey = appConfig.defaultVerificationServiceLink.apiKey ?? '';
+  let _credentialStatus = credentialStatus ? { ...credentialStatus } : null;
+
+  // issue credential status if not provided
+  if (!_credentialStatus) {
+    // issue credential status
+    _credentialStatus = await issueCredentialStatus({
+      host: new URL(vcKitAPIUrl).origin, // example: https://api.vc.example.com
+      apiKey,
+      bitstringStatusIssuer: issuer,
+    });
+  }
+
+  // issue vc
+  const body = constructCredentialObject({ context, type, issuer, credentialSubject, credentialStatus: _credentialStatus, ...restOfVC });
+  privateAPI.setBearerTokenAuthorizationHeaders(apiKey);
   const { verifiableCredential } = await privateAPI.post<VerifiableCredential>(
     `${vcKitAPIUrl}/credentials/issue`,
     body,
   );
   return verifiableCredential;
 };
 
+/**
+ * Issue credential status for the vc
+ * @param host - host for issuing the credential status
+ * @param apiKey - api key for issuing the credential status
+ * @param statusPurpose - purpose for the credential status
+ * @param bitstringStatusIssuer - issuer for the credential status
+ * @returns CredentialStatusReference
+ * 
+ * @example
+ * const host = 'https://api.vc.example.com';
+ * const apiKey = 'api-key';
+ * const statusPurpose = 'revocation';
+ * const bitstringStatusIssuer = 'did:example:123';
+ * const credentialStatus = await issueCredentialStatus({ host, statusPurpose, bitstringStatusIssuer });
+ */
+export const issueCredentialStatus = async (args: IArgIssueCredentialStatus): Promise<CredentialStatusReference> => {
+  const { host, apiKey, statusPurpose = 'revocation', ...rest } = args;
+  const body = { statusPurpose, ...rest };
+
+  // issue credential status
+  privateAPI.setBearerTokenAuthorizationHeaders(apiKey);
+  const response = await privateAPI.post<CredentialStatusReference>(`${host}/agent/issueBitstringStatusList`, body);
```

Comment:
@namhoang1604, should this endpoint be set in the config? Correct me if I'm wrong, but if an implementor uses their own VC service that is compliant with the vc-api spec, this will fail, right?

Reply:
I think we should set it as the endpoint of the VC API spec in the service. It means the mock app is using the VC service that complies with VCDM spec and VC API spec.